moped-editor/src/utils/projectComponentHelpers.js

```diff
@@ -0,0 +1,150 @@
+import { useEffect, useState } from "react";
```

Comment:
there's obvious overlap between this helper and `mapHelpers`, but i thought these component attribute-munging hooks deserved their own module.

Reply:
these were mostly extract from `ProjectComponentEdit.js` with minimal modification.